src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,116 @@
+package wolox.training.controllers;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.mockito.ArgumentMatchers.any;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.net.URI;
+import org.junit.Ignore;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+
+
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private BookRepository mockedBookRepository;
+
+    @MockBean
+    private Book theMist;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp() {
+        theMist = new Book("Terror", "Stephen King", "The Mist", "no value",
+            "SOME PUBLISHER", "2000", 123, "143565786", "imageOfBook");
+    }
+
+    @Test
+    public void givenBook_whenGetBookByAuthor_thenReturnsCorrectBook() throws Exception {
+        given(mockedBookRepository.findFirstByAuthor("The Mist")).willReturn(
+            java.util.Optional.ofNullable(theMist));
+
+        mockMvc.perform(get("/api/books/author/" + theMist.getTitle()))
```

Comment:
You could create a constant "/api/books" and concat this url with endpoint that is necessary.
For example: Constant + "author/" + theMist.getTitle()

or better, create a constant for each endpoint, so you should be concat only the IDs